### PR TITLE
System test cleanup

### PR DESCRIPTION
**Pull request by adamsj-ros**
# Description

All changes on `system_test_cleanup` have been previously reviewed but not all together. This PR is to do a quick review before merging to master.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] CI test
- [ ] local test

# Checklist:

- [ ] I have performed a self-review of my own code
- [x] New and existing system tests pass locally with my changes (when applicable)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: rmw_opendds_cpp/src/wait.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#include <rmw_opendds_cpp/condition_error.hpp>
+#include <rmw_opendds_cpp/event_converter.hpp>
+
+#include <rmw/error_handling.h>
+#include <rmw/impl/cpp/macros.hpp>
+#include <rmw/types.h>
+
+rmw_ret_t __handle_active_event_conditions(rmw_events_t* events)
+{
+  if (events) {
```
Comment (adamsj-ros) [inline rmw_opendds_cpp/src/wait.cpp:26]: If null, shouldn't this at least return an error or print something?

File: rmw_opendds_cpp/src/rmw_server.cpp [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  if (service) {
+    auto dds_server = static_cast<DDSServer*>(service->data);
+    if (dds_server) {
+      DDSServer::Raf::destroy(dds_server);
+      service->data = nullptr;
+    }
+    rmw_service_free(service);
+  }
+*/
+}
```
Comment (adamsj-ros) [inline rmw_opendds_cpp/src/rmw_server.cpp]: I suppose we should add a TODO here.

File: rmw_opendds_cpp/include/rmw_opendds_cpp/DDSEntity.hpp [hunk 3]
```diff
@@ -1,5 +1,3 @@
-// Copyright 2017 Open Source Robotics Foundation, Inc.
```
Comment (lij-oci) [inline rmw_opendds_cpp/include/rmw_opendds_cpp/DDSEntity.hpp:1]: How did these two files get related like this?

File: rmw_opendds_cpp/include/rmw_opendds_cpp/guard_condition.hpp [hunk 4]
```diff
@@ -12,13 +12,19 @@
 // See the License for the specific language governing permissions and
```
Comment (lij-oci) [inline rmw_opendds_cpp/include/rmw_opendds_cpp/guard_condition.hpp:12]: How did these two files get related like this?

**Review by lij-oci — APPROVED** [review]

File: rmw_opendds_cpp/src/wait.cpp (hunk 1 above)
Comment (lij-oci) [reply rmw_opendds_cpp/src/wait.cpp:26]: https://github.com/oci-labs/rmw_opendds/issues/64

**Pull request merged**